internal/api/controllers/private/highlevelJobRequest.go

```diff
@@ -0,0 +1,109 @@
+	var input RunInputV2List
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	for _, run := range input {
+		requestType, err := validateJobRequestFields(run)
+		if err != nil {
+			instrumentation.InvalidHighLevelJobRequest(ctx, requestType, err)
+			return invalidRequest(ctx, err)
+		}
+	}
+
+	// processing individual requests concurrently
+	result := input.PMapRunCreated(func(runInputV2 RunInputV2) *RunCreated {
+		requestTypeLabel := getRequestTypeLabel(runInputV2)
+		context := utils.WithOrgId(ctx.Request().Context(), string(runInputV2.OrgId))
+		context = utils.WithRequestType(context, requestTypeLabel)
+
+		recipient := parseValidatedUUID(string(runInputV2.Recipient))
+
+		hosts := parseRunHosts(runInputV2.Hosts)
+
+		var parsedSatID *uuid.UUID
+		if runInputV2.RecipientConfig != nil && runInputV2.RecipientConfig.SatId != nil {
+			parsedSatID = utils.UUIDRef(parseValidatedUUID(string(*runInputV2.RecipientConfig.SatId)))
+		}
+
+		runInput := RunInputV2GenericMap(runInputV2, recipient, hosts, parsedSatID, this.config)
+
+		hostConnectorDetails, err := this.inventoryConnectorClient.GetHostConnectionDetails(
+			context,
+			extractInventoryIds(runInput.Hosts),
+			this.config.GetString("inventory.connector.ordered.how"),
+			this.config.GetString("inventory.connector.ordered.by"),
+			this.config.GetInt("inventory.connector.limit"),
+			this.config.GetInt("inventory.connector.offset"),
+		)
+		if err != nil {
+			utils.GetLogFromContext(context).Error(err)
+			return handleRunCreateError(err)
+		}
+
+		satellite, directConnected, noRhc := sortHostsByRecipient(hostConnectorDetails)
+
+		if len(satellite) > 1 || len(directConnected) > 1 || len(noRhc) > 1 {
+			instrumentation.InvalidHighLevelJobRequest(ctx, requestTypeLabel, fmt.Errorf("Multiple recipients for host list found. Provided recipient and host list mismatch."))
+			return &RunCreated{Code: http.StatusBadRequest}
+		}
+
+		// when RHC is not setup, do not dispatch playbook run request
+		if noRhc != nil {
+			noRHCRecipientConnInfo := []RecipientWithConnectionInfo{getRHCStatus(noRhc, OrgId(runInput.OrgId))}
+			instrumentation.InvalidHighLevelJobRequest(ctx, requestTypeLabel, fmt.Errorf("RHC not installed. Recipient: %s", noRHCRecipientConnInfo[0].Recipient))
+			return runCreateError(http.StatusNotFound)
+		}
+
+		if satellite != nil {
+			satelliteStatus, err := getSatelliteStatus(ctx, this.cloudConnectorClient, this.sourcesConnectorClient, OrgId(runInput.OrgId), satellite)
+			if err != nil {
+				utils.GetLogFromContext(context).Errorf("Error retrieving Satellite status: %s", err)
+				return runCreateError(http.StatusNotFound)
+			}
+			if satelliteStatus[0].Status != "connected" {
+				instrumentation.InvalidHighLevelJobRequest(ctx, requestTypeLabel, fmt.Errorf("Satellite not connected"))
+				return runCreateError(http.StatusNotFound)
+			}
```

Comment:
I think we need a for loop here.  I believe the `satellite` variable is an array of satellite recipient ids.  So we need to loop over those and send messages to each of those satellites.

```
for each satelliteId in satellite {
// get status of satellite

// build the run input
runInput := RunInputV2GenericMap(runInputV2, satelliteId, hosts, parsedSatID, this.config)    
    
// send the message
runID, _, err := this.dispatchManager.ProcessRun(context, runInput.OrgId, middleware.GetPSKPrincipal(context), runInput )

// buffer up the runID 
}
```

We can't return here on an error as there could be directly connected hosts that we need to send messages to.  We'll need to buffer the error and continue processing.